apps/vs-code-designer/src/app/commands/deploy/hybridLogicApp/index.ts

```diff
@@ -24,44 +28,122 @@ export const deployHybridLogicApp = async (context: IActionContext, node: SlotTr
-          storageName: smbFolderName,
+          storageName: newSmbFolderName,
           subscriptionId: node.subscription.subscriptionId,
-          resourceGroup: node.resourceGroupName,
+          resourceGroup: node.resourceGroupName ?? node.hybridSite.id.split('/')[4],
           siteName: node.hybridSite.name,
+          hybridApp: node.hybridSite,
         };
-        await createHybridApp(context, accessToken, hybridAppOptions);
-
-        progress.report({ increment: 16, message: 'Deleting SMB folder' });
-        await deleteSMBFolder(mountDrive, smbFolderName);
+        await createOrUpdateHybridApp(context, accessToken, hybridAppOptions);
 
-        progress.report({ increment: 16, message: 'Unmounting SMB' });
-        await unMountSMB(mountDrive);
+        if (currentSmbFolder) {
+          progress.report({ increment: 16, message: 'Cleaning up previous resources' });
+          await cleanSMB(node.hybridSite.environmentId, currentSmbFolder, accessToken);
+          await deleteSMBFolder(mountDrive, currentSmbFolder);
+        }
 
-        progress.report({ increment: 20, message: 'Cleaning SMB' });
-        await cleanSMB(node, accessToken);
+        progress.report({ increment: 20, message: 'Unmounting SMB' });
       }
     );
   } catch (error) {
     throw new Error(`${localize('errorDeployingHybridLogicApp', 'Error deploying hybrid logic app')} - ${error.message}`);
+  } finally {
+    await unMountSMB(mountDrive);
+  }
+};
+
+const getSMBDetails = async (context: IActionContext, node: SlotTreeItem) => {
+  const volumeMount = node.hybridSite.template.containers[0]?.volumeMounts?.find((v) => v.mountPath === '/home/site/wwwroot');
+  const smbVolume = node.hybridSite.template.volumes.find((v) => v.name === volumeMount.volumeName);
+
+  if (smbVolume.storageType !== 'Smb') {
+    throw new Error(localize('errorDeployingHybridLogicApp', `The logic app ${node.hybridSite.name} is not configured to use SMB`));
+  }
+  await getStorageInfoForConnectedEnv(node.hybridSite.environmentId, smbVolume.storageName, context, node);
+};
+
+const getStorageInfoForConnectedEnv = async (connectedEnvId: string, storageName: string, context: IActionContext, node: SlotTreeItem) => {
+  const credentials: ServiceClientCredentials | undefined = await getAccountCredentials();
+  const accessToken = await getAuthorizationToken(credentials);
+
+  const url = `${azurePublicBaseUrl}/${connectedEnvId}/storages/${storageName}?api-version=2024-02-02-preview`;
+
+  try {
+    const response = await axios.get(url, { headers: { authorization: accessToken } });
+
+    if (!isSuccessResponse(response.status)) {
+      throw new Error(response.statusText);
+    }
+
+    node.fileShare = {
+      hostName: response.data?.properties?.smb?.host,
+      path: response.data?.properties?.smb?.shareName,
+    };
+    const fileSharePath = `\\\\${node.fileShare.hostName}\\${node.fileShare.path.split(path.sep)[0]}`;
```

Comment:
the `\\\\` is really specific for windows based, you can use path.sep

Reply:
Yes, so this is really the cleanup step after the deployment is succesful, so we don't want the user to think the deployment failed just because the cleanup failed